python/ray/data/preprocessors/batch_mapper.py

```diff
@@ -43,9 +44,28 @@ class BatchMapper(Preprocessor):
 
     _is_fittable = False
 
-    def __init__(self, fn: Callable[["pandas.DataFrame"], "pandas.DataFrame"]):
+    def __init__(
+        self,
+        fn: Callable[
+            [Union["pandas.DataFrame", np.ndarray, Dict[str, np.ndarray]]],
+            Union["pandas.DataFrame", np.ndarray, Dict[str, np.ndarray]],
+        ],
+        batch_format="pandas",
+        # TODO: We should reach consistency of args between BatchMapper and map_batches.
+    ):
+        assert batch_format in [
+            "pandas",
+            "numpy",
+        ], "BatchMapper only supports pandas and numpy batch format."
```

Comment:
This should be raised as a `ValueError` rather than an `AssertionError`, since this is a user-provided argument that we're validating.